### ISSUE TITLE
ical params should be case insensitive

### ISSUE THREAD

**Issue by rahearn** [icalendar/icalendar#116]
Currently, `ical_params` keys are case sensitive. This can lead to bugs with UTC DateTime formatting and other surprising behavior.

For example, with lowercase `tzid` this code is correct:

```
>> e = Icalendar::Event.new
>> e.dtstart = Icalendar::Values::DateTime.new "20150209T153000", 'tzid' => 'UTC'
>> e.to_ical
=> "BEGIN:VEVENT\r\nDTSTAMP:20150409T141700Z\r\nUID:a2d8c42a-e572-4b12-b7bd-df11c5589929\r\nDTSTART:20150209T153000Z\r\nEND:VEVENT\r\n"
```

While uppercase results in a bad format:

```
>> e = Icalendar::Event.new
>> e.dtstart = Icalendar::Values::DateTime.new "20150209T153000", 'TZID' => 'UTC'
>> e.to_ical
=> "BEGIN:VEVENT\r\nDTSTAMP:20150409T141700Z\r\nUID:a2d8c42a-e572-4b12-b7bd-df11c5589929\r\nDTSTART;TZID=UTC:20150209T153000\r\nEND:VEVENT\r\n"
```

**Comment by rahearn** [comment]
Fixed on master and will be released in 2.3.0 (later today?)

**Comment by worikgh** [comment]
On 27/04/15 07:26, Ryan Ahearn wrote:
> Closed #116.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/icalendar/icalendar/issues/116#event-290331162
> 

But do any of the test cases have mixed case?  I cannot see any

Worik

-- 
Why is the legal status of chardonnay different to that of cannabis?
       worik.stanton@gmail.com 021-1680650, (03) 4821804
                          Aotearoa (New Zealand)
                             I voted for love

**Comment by rahearn** [comment]
@worikgh https://github.com/icalendar/icalendar/blob/master/spec/downcased_hash_spec.rb tests the hash directly, while https://github.com/icalendar/icalendar/blob/master/spec/values/date_time_spec.rb#L59 tests that the specific symptom reported in issue #106 is now fixed
